pkg/driver/controller_modify_volume.go

```diff
@@ -107,55 +113,106 @@ func mergeModifyVolumeRequest(input modifyVolumeRequest, existing modifyVolumeRe
 		}
 		existing.modifyDiskOptions.VolumeType = input.modifyDiskOptions.VolumeType
 	}
-
+	if input.modifyTagsOptions != nil {
+		if len(input.modifyTagsOptions.TagsToAdd) > 0 || len(input.modifyTagsOptions.TagsToDelete) > 0 {
+			if existing.modifyTagsOptions != nil && input.modifyTagsOptions != existing.modifyTagsOptions {
+				return existing, fmt.Errorf("Different tags were requested by a previous request. Current: %v, Requested: %v", existing.modifyTagsOptions, input.modifyTagsOptions)
+			}
+			existing.modifyTagsOptions = &cloud.ModifyTagsOptions{
+				TagsToAdd:    input.modifyTagsOptions.TagsToAdd,
+				TagsToDelete: input.modifyTagsOptions.TagsToDelete,
+			}
+		}
+	}
 	return existing, nil
 }
 
+func executeModifyTagsRequest(volumeID string, options modifyVolumeRequest, c cloud.Cloud) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
+	defer cancel()
+	if options.modifyTagsOptions != nil {
```

Comment:
You can move the `options.modifyTagsOptions != nil` check to the beginning and that will avoid creating an unnecessary context when there are no tags to modify.

Also, curious to know why we are creating a new context here in the first place - is it just how its done elsewhere? should the caller provide an already existing context instead?  injecting the context as opposed to creating it inside a nested function is generally a good idea.

Reply:
Good catch, thank you, it was redundant to create the context here as it can just be passed in from the one already created in executeModifyVolumeRequest